manifesto

```diff
@@ -372,13 +399,97 @@ class CPortDescriptor(Descriptor):
         r += "protocol = {:#x}\n".format(self.protocol_num)
         return r
 
+class PropertyDescriptor(Descriptor):
+    def __init__(self, id_, name_stringid, typ, value, section):
+        super(PropertyDescriptor, self).__init__(section)
+        if id_ == 0:
+            raise Error("invalid id for '[{}]' (cannot be 0)".format(section))
+        self.id_ = id_
+        self.name_stringid = name_stringid
+        self.typ = typ
+        self.value = value
+        self._parent = None
+
+    @property
+    def parent(self):
+        return self._parent
+
+    @parent.setter
+    def parent(self, desc):
+        if self._parent is not None:
+            raise Error("multiple references to [{}]".format(self.section))
+        self.used = True
+        self._parent = desc
+
+    def __str__(self):
+        r = ""
+        r += "[{} {:#x}]\n".format(MnfsParser.PROPERTY_DESC, self.id_)
+        r += "name-string-id = {}\n".format(self.name_stringid)
+        r += "type = {}\n".format(self.typ)
+        r += "value = {}\n".format(self.value)
+        return r
+
+class DeviceDescriptor(Descriptor):
+    device_protocol = {
+            0x01: 'SPI',
+            0x02: 'I2C',
+            0x03: 'UART',
+            0x04: 'SPI_GPIO_CS',
```

Comment:
@cfriedt I don't think this needs to be handled right away, but by switching to CPort protocols, these additional protocols will be removed. These special cases are here to enable special driver modes. I believe moving them to properties or an alternative extension does make sense to keep CPort and DeviceDescriptor well in-sync.

Reply:
I think it would be helpful to have a clearer description about what these protocols are for. I'm a bit hesitant to give it a thumbs-up because it may be a problem that is already solved on the Greybus / controller side. I'm not sure if that was brought up on LKML.